pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -539,12 +540,13 @@ func (s *DiscoveryServer) edsUpdate(shard, serviceName string,
 				// Requires a CDS push and full sync.
 				adsLog.Infof("Endpoint updating service account %s %s", e.ServiceAccount, serviceName)
 				s.ConfigUpdate(true)
```

Comment:
This can be called multi-times in a loop.

Reply:
Since you understand this logic, can you update this either by adding a lot of comments or though some other reorg such that the flow is easier to understand?